.github/workflows/build_and_test.yml

```diff
@@ -20,14 +20,30 @@ concurrency:
 
 jobs:
 
+  get-torch-commit:
+    name: "Get torch commit"
+    uses: ./.github/workflows/_get_torch_commit.yml
+
   build-torch-xla:
     name: "Build PyTorch/XLA"
     uses: ./.github/workflows/_build_torch_xla.yml
+    needs: get-torch-commit
     with:
       dev-image: us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/development:3.10_tpuvm
+      torch-commit: ${{needs.get-torch-commit.outputs.torch_commit}}
     secrets:
       gcloud-service-key: ${{ secrets.GCLOUD_SERVICE_KEY }}
 
+  build-torch-with-cuda:
+    name: "Build PyTorch with CUDA"
+    uses: ./.github/workflows/_build_torch_with_cuda.yml
+    needs: get-torch-commit
+    with:
+      # note that to build a torch wheel with CUDA enabled, we do not need a GPU runner.
+      dev-image: us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/development:3.10_cuda_12.1
+      torch-commit: ${{needs.get-torch-commit.outputs.torch_commit}}
+      runner: linux.8xlarge.nvidia.gpu
```

Comment:
A non-GPU machine should work here, as long as you have the CUDA docker image.

Reply:
Yeah, I tried it on a non-GPU machine but it didn't work. I put the detail here https://github.com/pytorch/xla/pull/7140#issuecomment-2140757431.